components/org.wso2.carbon.identity.application.authz.xacml/src/test/java/org.wso2.carbon.identity.application.authz.xacml/handler.impl/XACMLBasedAuthorizationHandlerTest.java

```diff
@@ -0,0 +1,169 @@
+import static org.testng.AssertJUnit.assertEquals;
+
+/**
+ * XACMLBasedAuthorizationHandlerTest defines unit tests for XACMLBasedAuthorizationHandler class.
+ */
+@PrepareForTest({LogFactory.class, FrameworkUtils.class, PolicyCreatorUtil.class, PolicyBuilder.class})
+@PowerMockIgnore("javax.xml.*")
+public class XACMLBasedAuthorizationHandlerTest {
+
+    private XACMLBasedAuthorizationHandler xacmlBasedAuthorizationHandler;
+    private AuthenticationContext context;
+    private Log log = mock(Log.class);
+
+    private static final String ADMIN_USER = "admin_user";
+    private static final String DECISION = "decision";
+    private static final String RULE_EFFECT_PERMIT = "Permit";
+    private static final String RULE_EFFECT_NOT_APPLICABLE = "NotApplicable";
+    private static final String POLICY = "policy";
+    private static final String ERROR = "error";
+    private static String XACML_RESPONCE = "<ns:root xmlns:ns=\"urn:oasis:names:tc:xacml:3.0:core:schema:wd-17\">"
+            +   "<ns:Result>"
+            +   "<ns:Decision>"
+            +   DECISION
+            +   "</ns:Decision>"
+            +   "</ns:Result>"
+            +   "</ns:root>";
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new PowerMockObjectFactory();
+    }
+
+    @BeforeClass
+    public void init() {
+
+        mockStatic(LogFactory.class);
+        when(LogFactory.getLog(XACMLBasedAuthorizationHandler.class)).thenReturn(log);
+        xacmlBasedAuthorizationHandler = spy(XACMLBasedAuthorizationHandler.getInstance());
+        context = mock(AuthenticationContext.class);
+        when(log.isDebugEnabled()).thenReturn(true);
+    }
+
+    @Test
+    public void testGetInstance() {
+
+        XACMLBasedAuthorizationHandler xacmlBasedAuthorizationHandler = XACMLBasedAuthorizationHandler.getInstance();
+        assertNotNull(xacmlBasedAuthorizationHandler);
+    }
+
+    @Test
+    public void testCreateRequestDTO() throws Exception {
+
+        SequenceConfig sequenceConfig = mock(SequenceConfig.class);
+        AuthenticatedUser authenticatedUser = mock(AuthenticatedUser.class);
+        when(context.getSequenceConfig()).thenReturn(sequenceConfig);
+        when(sequenceConfig.getAuthenticatedUser()).thenReturn(authenticatedUser);
+        when(authenticatedUser.getUserName()).thenReturn(ADMIN_USER);
+        RequestDTO requestDTO = WhiteboxImpl.invokeMethod(xacmlBasedAuthorizationHandler,
+                "createRequestDTO", context);
+        assertTrue(requestDTO.getRowDTOs().size() == 8);
+    }
+
+    @Test
+    public void testEvaluateXACMLResponse() throws Exception {
+
+        String responce = WhiteboxImpl.invokeMethod(xacmlBasedAuthorizationHandler, "evaluateXACMLResponse",
+                XACML_RESPONCE);
+        assertEquals(responce, DECISION);
+    }
+
+    @Test
+    public void testIsAuthorized() throws Exception {
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
```

Comment:
How about defining this at class level and using @Mock annotation?

Reply:
I wonder any specific reason for declaring it at class level. This object only required in testing isAuthorized method.